src/websocket-connection.js

```diff
@@ -116,14 +116,20 @@ WebsocketConnection.prototype.close = function() {
     return Promise.resolve();
   }
   return new Promise(function(resolve) {
+    this._close();
+    resolve();
+  }.bind(this));
+};
+
+WebsocketConnection.prototype._close = function() {
+  if (!this.isClosed()) {
     if (this._webSocket.terminate) {
       this._webSocket.terminate();
     } else {
       this._webSocket.close();
     }
-    this.emit('close');
-    resolve();
-  }.bind(this));
+  }
+  this.emit('close');
```

Comment:
The connection can be already closed when calling this method. This method is used for the case when `.close()` is called by client and for the case when `._close()` is called by closed `_webSocket`.